examples/NFD/stateless_firewall/stateless_firewall.cpp

```diff
@@ -0,0 +1,311 @@
+
+/************************************************************************************
+* Filename:   stateful_firewall.cpp
+* Author:     Hongyi Huang(hhy17 AT mails.tsinghua.edu.cn), Bangwen Deng, Wenfei Wu
+* Copyright:
+* Disclaimer: This code is presented "as is" without any guarantees.
+* Details:    This code is the implementation of state_firewall from NFD project,
+              a C++ based NF developing framework designed by Wenfei's group
+              from IIIS, Tsinghua University, China.
+*************************************************************************************/
+
+#include <errno.h>
+#include <getopt.h>
+#include <inttypes.h>
+#include <stdarg.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/queue.h>
+#include <unistd.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <rte_common.h>
+#include <rte_ip.h>
+#include <rte_mbuf.h>
+
+#include "onvm_nflib.h"
+#include "onvm_pkt_helper.h"
+
+#ifdef __cplusplus
+}
+#endif
+
+// NFD ADD
+#include <pcap.h>
+#include <signal.h>
+#include <sys/time.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <cstring>
+#include <iomanip>
+#include <iostream>
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+#include "basic_classes.h"
+#include "decode.h"
+
+using namespace std;
+
+#define NF_TAG "stateless_firewall"
+
+/* Struct that contains information about this NF */
+struct onvm_nf_info *nf_info;
+
+/* number of package between each print */
+static uint32_t print_delay = 1000000;
+
+static uint32_t destination;
+
+/*******************************NFD features********************************/
+Flow::Flow(u_char *packet, int totallength) {
+        /*Decoding*/
+        this->pkt = packet;
+        int ethernet_header_length = 14;
+        /* Header lengths in bytes */
+        EtherHdr *e_hdr = (EtherHdr *)packet;
+        if (ntohs(e_hdr->ether_type) == 0x8100)
+                ethernet_header_length = 14 + 4; /* For 802.1Q Virtual LAN */
+        else
+                ethernet_header_length = 14; /* For general wired */
+
+        IPHdr *ip_hdr = (IPHdr *)(packet + ethernet_header_length);
+        int src_addr = ntohl(ip_hdr->ip_src.s_addr);
+        this->headers[Sip] = new IP(src_addr, 32);
+        short protocol = (short)(ip_hdr->ip_proto);
+        this->headers[Tcp] = new int((protocol == IPPROTO_TCP) ? 1 : 0);
+}
+void
+Flow::clean() {
+}
+
+long int _counter = 0;
+long int _drop = 0;
+unordered_map<string, int> F_Type::MAP = unordered_map<string, int>();
+unordered_map<string, int> F_Type::MAP2 = unordered_map<string, int>();
+int
+process(Flow &f);
+Flow f_glb;
+
+/////time////
+struct timeval begin_time;
+struct timeval end_time;
+
+IP ip1("41.177.117.184/32");
+
+void
+_init_() {
+        (new F_Type())->init();
+}
+
+int
+process(Flow &f) {
+        if (*((IP *)f.headers[Sip]) != ip1) {
+        }
+        // to be fixed
```

Comment:
Whats supposed to happen here?

Reply:
no operation I think. I will delete the comment.